### PR TITLE
feat: error handling/report in rust lib

### PR DESCRIPTION
**Pull request by sjeohp**
- add error handling so that error/exception messages can be returned across the bridge
- simplify C/Swift API, especially string allocation/destruction

### CODE REVIEW

File: rust/signer/src/lib.rs [hunk 1]
```diff
@@ -16,386 +16,424 @@
...
+		msg: &str,
+		signature: &str
+	) -> crate::Result<bool> {
+		let keypair = sr25519::KeyPair::from_suri(suri)
+			.ok_or(crate::Error::KeyPairIsNone)?;
+		let message: Vec<u8> = msg.from_hex()
+			.map_err(|e| crate::Error::FromHex(e))?;
+		let signature: Vec<u8> = signature.from_hex()
+			.map_err(|e| crate::Error::FromHex(e))?;
+		keypair.verify_signature(&message, &signature)
```
Comment (hanwencheng) [inline rust/signer/src/lib.rs:247]:
```suggestion
		Ok(keypair.verify_signature(&message, &signature).unwrap_or(false))
```

Comment (hanwencheng) [inline rust/signer/src/lib.rs:247]: or probabaly add a verify error with `ok_or`?

File: rust/signer/src/lib.rs [hunk 2]
```diff
@@ -16,386 +16,424 @@
...
+	#[test]
+	fn decrypt_with_ref() {
+		let decrypted_result = decrypt_data(ENCRYPTED_SEED, String::from(PIN)).unwrap();
+		assert_eq!(SEED_PHRASE, decrypted_result);
+	}
+
+	#[test]
+	fn test_generate_substrate_address() {
+		let data_pointer = decrypt_data_ref(ENCRYPTED_SEED, String::from(PIN)).unwrap();
+		let address = substrate_wallet_address_with_ref(data_pointer, SURI_SUFFIX, 42).unwrap();
```
Comment (hanwencheng) [inline rust/signer/src/lib.rs:427]:
```suggestion
		let address = substrate_address_with_ref(data_pointer, SURI_SUFFIX, 42).unwrap();
```

**Comment by hanwencheng** [comment]
Could you please explain the usage of file `jmod.rs.rustfmt` ?

**Comment by hanwencheng** [comment]
Also the `yarn test-rust` does not passed.

**Comment by sjeohp** [comment]
> Could you please explain the usage of file `jmod.rs.rustfmt` ?

No use, removed.

**Pull request merged**